src/ray/raylet/node_manager.cc

```diff
@@ -906,11 +909,7 @@ void NodeManager::ResourceCreateUpdated(const NodeID &node_id,
                                                         new_resource_capacity);
   }
   RAY_LOG(DEBUG) << "[ResourceCreateUpdated] Updated cluster_resource_map.";
-
-  if (node_id == self_node_id_) {
-    // The resource update is on the local node, check if we can reschedule tasks.
-    cluster_task_manager_->ScheduleAndDispatchTasks();
-  }
+  cluster_task_manager_->ScheduleAndDispatchTasks();
```

Comment:
So, when the local resources are updated, `ScheduleAndDispatchTasks ` is invoked? Can you point me where that happens?

Reply:
Local resources are updated by local code. https://github.com/ray-project/ray/blob/master/src/ray/raylet/node_manager.cc#L1634

Right now, only pg will change local resource